plugins/module_utils/mso.py

```diff
@@ -833,7 +833,8 @@ def lookup_site(self, site):
 
         s = self.get_obj("sites", name=site)
         if not s:
-            self.fail_json(msg="Site '%s' is not a valid site name." % site)
+            self.module.warn("Site '{0}' is not a valid site name.".format(site))
```

Comment:
How do we generically want to handle warnings?

Reply:
Added ignore_404 parameter to the lookup methods. Based on the ignore_404 value will fail the lookup method.